src/main/java/br/com/sbk/sbking/networking/server/Table.java

```diff
@@ -12,43 +12,50 @@
 import br.com.sbk.sbking.core.Player;
 import br.com.sbk.sbking.networking.server.gameServer.GameServer;
 
+/**
+ * This class has two responsibilities: 1: control the players and the
+ * directions they are sitting 2: host a gameServer. This way, the gameServer
+ * should remain isolated from the lower layers and only be altered via this
+ * table.
+ */
 public class Table {
 
-  private Map<Direction, ClientGameSocket> playerSockets;
-  private Collection<ClientGameSocket> spectatorSockets;
-  private Map<UUID, ClientGameSocket> allSockets;
+  private Map<Direction, Player> seatedPlayers;
+  private Collection<Player> spectatorPlayers;
+  private Map<UUID, Player> allPlayers;
   private GameServer gameServer;
 
   public Table(GameServer gameServer) {
     this.gameServer = gameServer;
-    this.playerSockets = new HashMap<Direction, ClientGameSocket>();
-    this.spectatorSockets = new ArrayList<ClientGameSocket>();
-    this.allSockets = new HashMap<UUID, ClientGameSocket>();
+    this.gameServer.setTable(this);
+    this.seatedPlayers = new HashMap<Direction, Player>();
+    this.spectatorPlayers = new ArrayList<Player>();
+    this.allPlayers = new HashMap<UUID, Player>();
```

Comment:
If the Player has an UUID attribute, why we are mapping the information that we already know?

Reply:
maybe it should be a List

Reply:
The UUID attribute of a player gives me the UUID when I **already have** the Player. In this case, I have the opposite. I have an UUID and want to know which player has that UUID. That is why we need a Map. A List would also work, but it is slower (`O(n)` vs `O(1)`).

In this case, the `Table` is doing the interface between the network layer (which knows only `UUID`s) and the `core` which uses `Player`s.

While I was doing this task, I imagined a scenario where the `Table` would be part of `core` and would use only `Player` but there is still the need of having a class that knows how to make the translation. This class will probably appear still in this sprint.